integration/spadev0/util.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/data-preservation-programs/RetrievalBot/pkg/convert"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/env"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/model"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/requesterror"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/resolver"
+	"github.com/data-preservation-programs/RetrievalBot/pkg/task"
+	"github.com/pkg/errors"
+	"github.com/rjNemo/underscore"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type GroupID struct {
+	Provider string `bson:"provider"`
+	PieceCID string `bson:"piece_cid"`
+}
+type Row struct {
+	ID       GroupID         `bson:"_id"`
+	Document model.DealState `bson:"document"`
+}
+
+func AddSpadeTasks(ctx context.Context, requester string, replicasToTest map[int][]Replica) error {
+	// Connect to the database
+	stateMarketDealsClient, err := mongo.
```

Comment:
I don't think you really need statemarketdeals. 
It seems you are trying to find the client and deal_id, but those are not required information to perform retrieval testing